x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -339,6 +424,28 @@ export const StepDetailsForm: FC<Props> = React.memo(
               data-test-subj="transformCreateIndexPatternSwitch"
             />
           </EuiFormRow>
+          {createIndexPattern && !indexPatternTitleExists && previewDateColumns.length > 0 && (
```

Comment:
This component is already pretty massive. Please extract this control to the dedicated component

Reply:
Refactored in https://github.com/elastic/kibana/commit/3fe458e73e36261939a364c1a1101626b85ba7b1